internal/api/v1beta1/domain.go

```diff
@@ -0,0 +1,224 @@
+package v1beta1
+
+import (
+	"context"
+
+	grpczap "github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"github.com/raystack/frontier/core/domain"
+	"github.com/raystack/frontier/core/organization"
+	frontierv1beta1 "github.com/raystack/frontier/proto/v1beta1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+var (
+	grpcDomainNotFoundErr      = status.Errorf(codes.NotFound, "domain whitelist request doesn't exist")
+	grpcDomainAlreadyExistsErr = status.Errorf(codes.AlreadyExists, "domain name already exists for that organization")
+	grpcInvalidHostErr         = status.Errorf(codes.NotFound, "invalid domain. No such host found")
+	grpcTXTRecordNotFound      = status.Errorf(codes.NotFound, "required TXT record not found for domain verification")
+	grpcDomainMisMatchErr      = status.Errorf(codes.InvalidArgument, "user and org's whitelisted domains doesn't match")
+)
+
+type DomainService interface {
+	Get(ctx context.Context, id string) (domain.Domain, error)
+	List(ctx context.Context, flt domain.Filter) ([]domain.Domain, error)
+	ListOrgByDomain(ctx context.Context, domain string) ([]string, error)
+	Delete(ctx context.Context, id string) error
+	Create(ctx context.Context, toCreate domain.Domain) (domain.Domain, error)
+	VerifyDomain(ctx context.Context, id string) (domain.Domain, error)
+	Join(ctx context.Context, orgID string, userID string) error
+	InitDomainVerification(ctx context.Context) error
+	Close()
+}
+
+func (h Handler) AddOrganizationDomain(ctx context.Context, request *frontierv1beta1.AddOrganizationDomainRequest) (*frontierv1beta1.AddOrganizationDomainResponse, error) {
+	logger := grpczap.Extract(ctx)
+
+	if request.GetOrgId() == "" || request.GetDomain() == "" {
+		return nil, grpcBadBodyError
+	}
```

Comment:
Do we have these checks in proto as well?

Reply:
yes